client/landing/jetpack-cloud/sections/scan/main.tsx

```diff
@@ -199,7 +199,7 @@ class ScanPage extends Component< Props > {
 
 	renderScanState() {
 		const { site, scanState } = this.props;
-		if ( ! scanState ) {
+		if ( ! scanState || ! site ) {
```

Comment:
I wonder if there is a way to tell TS that this cannot happen. My understanding is that you can't reach this page unless you have selected a site, therefore, site shouldn't be undefined. Am I right?

Reply:
Found something! https://medium.com/better-programming/cleaner-typescript-with-the-non-null-assertion-operator-300789388376

Will do a quick refactor to test :)

Reply:
Looks like that is forbidden by ESLint. I'll merge once the tests pass and if we find a better solution we can refactor later.